### ISSUE TITLE
OSX Failed to launch delve with this error Error: spawn /usr/local/bin/dlv ENOENT

### ISSUE THREAD

**Issue by jkhelil** [Microsoft/vscode-go#466]
I ma using vscode 1.4.0, and trying to launch delve for go debugging
I get this log from  the console 
messageService.ts:119Failed to continue: "Error: spawn /usr/local/bin/dlv ENOENT": Error: Failed to continue: "Error: spawn /usr/local/bin/dlv ENOENT"
    at Object.C [as create](file:////Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:34:19570)
    at file:////Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:105:21677
    at Object.m [as _notify](file:////Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:30:8153)
    at Object.B.enter (file:////Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:30:11658)
    at i.Class.define._run (file:////Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:30:12979)
    at i.Class.define._chainedError (file:////Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:30:12338)
    at i (file:////Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:30:10161)
    at i.Class.define.then (file:////Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:30:13788)
    at Object.P.enter (file:////Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:30:10230)
    at i.Class.define._run (file:////Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:30:12979)

Any help would be greatly appreciated. Thanks!

**Comment by deejross** [comment]
The only time I've ever seen this error is when VSCode was opened in something other than their $GOPATH folder (workspace), or the configuration in $GOPATH/.vscode/launch.json points to the wrong folder.

**Comment by ramya-rao-a** [comment]
@jkhelil Can you check if the discussion in #332 helps? It arrives at the same conclusion as @deejross

**Comment by ramya-rao-a** [comment]
@jkhelil @mrichman Do you still see this issue or fixing `launch.json` has fixed this for you?

**Comment by mrichman** [comment]
I had to replace `${workspaceroot}` with the absolute path to my project. In this case, `"program": "/Users/mrichman/go/src/github.com/mrichman/myapp/cmd/myapp/main.go"`

**Comment by ramya-rao-a** [comment]
@mrichman You can use `${workspaceroot}` instead of typing out the whole absolute path as well.
For example in your case you can set 

`"program": "${workspaceroot}/cmd/myapp/main.go"`

instead of 

`"program": "/Users/mrichman/go/src/github.com/mrichman/myapp/cmd/myapp/main.go"`

Assuming that you have opened /Users/mrichman/go/src/github.com/mrichman/myapp in VS Code

**Comment by ramya-rao-a** [comment]
Closing this as we havent heard back anything recently.

**Comment by eureka-gh** [comment]
I have the same issue even if with all above suggestions.

**Comment by ramya-rao-a** [comment]
@eureka-gh Delve errors out with`could not launch process: could not fork/exec` if there are issues with code signing/certificates.

See https://github.com/Microsoft/vscode-go/wiki/Debugging-Go-code-using-VS-Code#could-not-launch-process-could-not-forkexec

**Comment by ramya-rao-a** [comment]
@mrichman @eureka-gh Another thing I just noticed, there is a casing issue. `${workspaceroot}` won't work. `${workspaceRoot}` is the right token to use.

**Comment by eureka-gh** [comment]
Hi ramya, Thanks for your prompt response to my question.
I followed a couple of solution mentioned in provided link. New error's coming : (
- The mentioned "could not fork/exec" issue can be partially resolved by using sudo in CLI, how to launch the dlv debugger in VS code with sudo ?
- Even with "sudo dlv debug [fullpath to go file]" in CLI, this is what I've got 
```
bash-3.2$ sudo dlv debug [my personal path]/main.go
Password:
[my personal path]/routers.go:7:2: cannot find package "github.com/gorilla/mux" in any of:
        /usr/local/Cellar/go/1.7.5/libexec/src/github.com/gorilla/mux (from $GOROOT)
        ($GOPATH not set)
exit status 1
```
Actually, my $GOPATH is set, since echo $GOPATH give the right location. My dlv is the latest version.

Not so sure, if you can give me other hints on that ?

Many thanks!

**Comment by ramya-rao-a** [comment]
Where did you get the suggestion to use sudo? None of the linked items #717, #269 or https://github.com/derekparker/delve/issues/357 suggest using sudo.

"could not fork/exec" is always associated with code signing issues.

All the linked items talk about clean installing delve again

**Comment by eureka-gh** [comment]
@ramya-rao-a Sorry for not going though all your resources. I opened two many searching pages and totally lost. Now I am able to use dlv to debug my go code in VS code. I removed previous dlv which was installed by "go get " and reinstall new dlv with "brew install go-delve/delve/delve" which is pointed out at https://github.com/derekparker/delve/blob/master/Documentation/installation/osx/install.md

It turns out "github.com/derekparker/delve" is NOT a good way to install dlv on OSX.

Thanks for your support. It's very helpful.
